File: sdks/python/src/echo_agent_sdk/subagent_context_values.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class SubagentContext:
    tool_definitions: tuple[object, ...] = ()
    messages: tuple[object, ...] = ()
    store_present: bool = False
    parent_goal: str | None = None
    allowed_tools: tuple[str, ...] | None = None

    @classmethod
    def empty(cls) -> SubagentContext:
        return cls()

    def has_content(self) -> bool:
        return bool(
            self.tool_definitions
            or self.messages
            or self.store_present
            or self.parent_goal is not None
            or self.allowed_tools is not None
        )

File: sdks/typescript/src/subagent_context_values.ts
/** Subagent context snapshot values without owning tools, messages, or stores. */
export class SubagentContext {
  public constructor(
    public readonly toolDefinitions: readonly unknown[] = [],
    public readonly messages: readonly unknown[] = [],
    public readonly storePresent: boolean = false,
    public readonly parentGoal: string | null = null,
    public readonly allowedTools: readonly string[] | null = null,
  ) {}

  public static empty(): SubagentContext {
    return new SubagentContext();
  }

  public hasContent(): boolean {
    return this.toolDefinitions.length > 0
      || this.messages.length > 0
      || this.storePresent
      || this.parentGoal !== null
      || this.allowedTools !== null;
  }
}
